root/static/scripts/edit/URLCleanup.js

```diff
@@ -3478,7 +3481,63 @@ validationRules[LINK_TYPES.discographyentry.release] = function (url) {
       result: false,
     };
   }
-  return originalRule(url);
+  if (/^(https?:\/\/)?([^.\/]+\.)?wikidata\.org\//.test(url)) {
+    return {
+      error: l(
+        `Wikidata is not a discography entry. Please add this Wikidata link
+         to the release group instead.`,
+      ),
+      result: false,
+    };
+  }
+  return discographyRule(url);
+};
+
+// avoid Wikipedia/Wikidata being added as release-level license entry
+const licenseRule = validationRules[LINK_TYPES.license.release];
+validationRules[LINK_TYPES.license.release] = function (url) {
```

Comment:
Not sure I understand why we override the functions instead of adding it into the originals (or alternatively, call the added functions from the original function instead of doing it backwards).

Reply:
Maybe it wasn’t possible in the past, but since `LINK_TYPES.wikipedia` has no `release` key, it should be possible to auto-select `discographyentry` type for release using the _wikipedia_ rule and invalidate it there. Same for wikidata.

Reply:
Want to add a ticket for the refactoring?

Reply:
@mwiencek: A rule can possibly auto-select only one relationship type for each entity type. And rules have to be exclusive to each other. The _original function_ is created dynamically by testing all rules, so we can only modify functions like that.

@reosarevok: Actually, we cannot just _auto-select `discographyentry` type for release using the wikipedia rule and invalidate it there_ because editor would still be able to select another relationship type from the release editor.

This is more or less related to MBS-9902 which requires a deep refactoring of URL cleanup code. Possibly an idea for GSoC along with reworking URL relationship editor UI?